docs/book/pattern/class-cache.md

```diff
@@ -8,34 +8,65 @@ class being cached, and caches static properties.
 ## Quick Start
 
 ```php
-use Laminas\Cache\PatternFactory;
-
-$classCache = PatternFactory::factory('class', [
-    'class'   => 'MyClass',
-    'storage' => 'apc',
-]);
+use Laminas\Cache\Pattern\ClassCache;
+use Laminas\Cache\Pattern\PatternOptions;
+
+$classCache = new ClassCache(
+    $storage,
+    new PatternOptions([
+        'class' => 'MyClass',
+    ])
+);
 ```
 
+> ### Storage Adapter
+>
+> The `$storage` adapter can be any adapter which implements the `StorageInterface`. Check out the [Pattern Quick Start](./intro.md#quick-start)-Section for a standard adapter which can be used here.
+
 ## Configuration Options
 
 Option | Data Type | Default Value | Description
 ------ | --------- | ------------- | -----------
-`storage` | `string | array | Laminas\Cache\Storage\StorageInterface` | none | Adapter used for reading and writing cached data.
+`storage` | `string\|array\|Laminas\Cache\Storage\StorageInterface` | none | **deprecated** Adapter used for reading and writing cached data.
 `class` | `string` | none | Name of the class for which to cache method output.
-`cache_output` | `boolean` | `true` | Whether or not to cache method output.
-`cache_by_default` | `boolean` | `true` | Cache all method calls by default.
+`cache_output` | `bool` | `true` | Whether or not to cache method output.
+`cache_by_default` | `bool` | `true` | Cache all method calls by default.
 `class_cache_methods` | `array` | `[]` | List of methods to cache (if `cache_by_default` is disabled).
 `class_non_cache_methods` | `array` | `[]` | List of methods to omit from caching (if `cache_by_default` is enabled).
 
+## Examples
+
+### Caching of Import Feeds
+
+```php
+use Laminas\Cache\Pattern\ClassCache;
+use Laminas\Cache\Pattern\PatternOptions;
+
+$cachedFeedReader = new ClassCache(
+    $storage,
+    new PatternOptions([
+        'class' => \Laminas\Feed\Reader\Reader::class,
+        
+        // The feed reader doesn't output anything,
+        // so the output doesn't need to be caught and cached:
+        'cache_output' => false,
+    ])
+);
+
+$feed = $cachedFeedReader->call("import", array('http://www.planet-php.net/rdf/'));
+
+// OR
+$feed = $cachedFeedReader->import('http://www.planet-php.net/rdf/');
+```
```

Comment:
```suggestion
```php
use Laminas\Cache\Pattern\ClassCache;
use Laminas\Cache\Pattern\PatternOptions;
use Laminas\Feed\Reader\Reader;

$cachedFeedReader = new ClassCache(
    $storage,
    new PatternOptions([
        'class' => Reader::class,
        
        // The feed reader doesn't output anything,
        // so the output doesn't need to be caught and cached:
        'cache_output' => false,
    ])
);

$feed = $cachedFeedReader->call("import", ['https://github.com/laminas/laminas-cache/releases.atom']);
// or
$feed = $cachedFeedReader->import('https://github.com/laminas/laminas-cache/releases.atom');
```
```